packages/angular_devkit/core/src/jobs/README.md

```diff
@@ -0,0 +1,501 @@
+interface Scheduler {
+  /**
+   * Pause the scheduler, temporary queueing _new_ jobs. Returns a resume function that should be
+   * used to resume execution. If multiple `pause()` were called, all their resume functions must
+   * be called before the Scheduler actually starts new jobs. Additional calls to the same resume
+   * function will have no effect.
+   *
+   * Jobs already running are NOT paused. This is pausing the scheduler only.
+   *
+   * @returns A function that can be run to resume the scheduler. If multiple `pause()` calls
+   *          were made, all their return function must be called (in any order) before the
+   *          scheduler can resume.
+   */
+  pause(): () => void;
+}
+```
+
+## <a name="Dependencies"></a>Synchronizing and Dependencies
+When scheduling jobs, it is often necessary to run jobs after certain other jobs are finished. 
+This is done through the `dependencies` options in the `schedule()` method.
+
+These jobs will also be passed to the job being scheduled, through its context. This can be
+useful if, for example, the output of those jobs are of a known type, or have known side channels.
+
+An example of this would be a compiler that needs to know the output directory of other compilers
+before it, in a tool chain.
+
+### Dependencies
+When scheduling jobs, the user can add a `dependencies` field to the scheduling options. The
+scheduler will wait for those dependencies to finish before running the job, and pass those jobs 
+in the context of the job.
+
+### Accessing Dependencies
+Jobs are called with a `JobHandlerContext` as a second argument, which contains a
+`dependencies: Job<JsonValue>[]` member which contains all dependencies that were used when 
+scheduling the job. Those aren't fully typed as they are determined by the user, and not the job
+itself. They also can contain jobs that are not finished, and the job should use the `state`
+member of the job itself before trying to access its content.
+
+### Scheduler Sub Jobs
+The `JobHandlerContext` also contains a `scheduler` member which can be used to schedule jobs
+using the same scheduler that was used for the job. This allows jobs to call other jobs
+and wait for them to end.
+
+## Available Schedulers
+The Core Angular DevKit library provides 2 implementations for the `Scheduler` interface:
+
+## SimpleJobRegistry
+Available in the jobs namespace. A registry that accept job registration, and can also schedule
+jobs.
+
+```typescript
+import { jobs } from '@angular-devkit/core';
+
+const add = jobs.createJobHandler<number[], number>(
+  input => input.reduce((total, curr) => total + curr, 0),
+);
+
+// Register the job in a SimpleJobRegistry. Different registries have different API.
+const registry = new jobs.SimpleJobRegistry();
+registry.register(add, {
+  name: 'add',
+  input: { type: 'array', items: { type: 'number' } },
+  output: { type: 'number' },
+});
+
+registry.schedule('add', [1, 2, 3, 4]);
+```
+
+### Job Compatibility
```

Comment:
I don't think this section applies anymore since implements` doesn't exist.

Reply:
removed.